### PR TITLE
Allow deprecation warnings to bubble up

### LINKED ISSUE #1680: Consider suppressing deprecation warnings in `expect_warning(, NA)`
Since this is causing revdep failures

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1680

### CODE REVIEW

File: R/expect-no-condition.R [hunk 1]
```diff
@@ -93,13 +93,18 @@ expect_no_ <- function(base_class,
...
+  ignore_deprec <- base_class == "warning" && is.null(regexp) && is.null(class)
+
   capture <- function(code) {
     try_fetch(
       code,
       !!base_class := function(cnd) {
         if (!matcher(cnd)) {
           return(zap())
         }
+        if (!ignore_deprec && is_deprecation(cnd)) {
```
Comment (lionel-) [inline R/expect-no-condition.R:105]: Note: I find it hard to follow the logic here and how it interacts with the one in `cnd_matcher()`.

**Review by lionel- — APPROVED** [review]

File: R/expect-condition.R [hunk 2]
```diff
@@ -305,6 +311,10 @@ cnd_matcher <- function(class, pattern = NULL, ..., inherit = TRUE) {
   }
 }
 
+is_deprecation <- function(x) {
+  inherits(x, "lifecycle_warning_deprecated")
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/expect-condition.R:315]: Hmmm, or is that likely to be annoying to R core? i.e. if they deprecate a function, it'll now guaranteed to be swallowed by testthat.

File: R/expect-no-condition.R (hunk 1 above)
Comment (hadley) [reply R/expect-no-condition.R:105]: Oops, because I accidentally implemented it twice.

File: R/expect-condition.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (lionel-) [reply R/expect-condition.R:315]: I thought this lets the deprecation warnings bubble up?

**Comment by hadley** [comment]
I worked through the revdep failures, but I can't reproduce them locally and can't generally see why they would have changed since the last run. I'll re-run these broken packages in the hope they were just stochastic failures.

**Comment by hadley** [comment]
I re-ran just the broken packages and the unexpected problems went away.
